backend/src/sms/routes/tests/sms-campaign.routes.test.ts

```diff
@@ -32,7 +26,119 @@ afterAll(async () => {
+const mockGetSecretValue = jest.fn((_) => TWILIO_SECRET_VALUE)
+
+// Setup Mock AWS SecretsManager
+jest.mock('aws-sdk', () => {
+  return {
+    ...jest.requireActual('aws-sdk'),
+    SecretsManager: function () {
+      return {
+        getSecretValue: ({ SecretId }: { SecretId: string }) => ({
+          promise: () => mockGetSecretValue(SecretId),
+        }),
+      }
+    },
+  }
+})
+
+// Creates demo/non-demo campaign based on parameters
+const createCampaign = async ({ isDemo }: { isDemo: boolean }) =>
+  await Campaign.create({
+    name: 'Test Campaign',
+    userId: 1,
+    type: ChannelType.SMS,
+    protect: false,
+    valid: false,
+    demoMessageLimit: isDemo ? 20 : null,
+  })
+
+describe('POST /credentials', () => {
+  test('Non-Demo campaign should not be able to use demo credentials', async () => {
+    const nonDemoCampaign = await createCampaign({ isDemo: false })
+
+    const res = await request(app)
+      .post(`/campaign/${nonDemoCampaign.id}/sms/credentials`)
+      .send({
+        label: DefaultCredentialName.SMS,
+        recipient: '98765432',
+      })
+
+    expect(res.status).toBe(400)
+    expect(res.body).toEqual({
+      message: `Campaign cannot use demo credentials. ${DefaultCredentialName.SMS} is not allowed.`,
+    })
+
+    // SecretManager should not be called
+    expect(mockGetSecretValue).not.toHaveBeenCalled()
+  })
+
+  test('Demo Campaign should not be able to use non-demo credentials', async () => {
+    const demoCampaign = await createCampaign({ isDemo: true })
+
+    const NON_DEMO_CREDENTIAL_LABEL = 'Some Credential'
+
+    const res = await request(app)
+      .post(`/campaign/${demoCampaign.id}/sms/credentials`)
+      .send({
+        label: NON_DEMO_CREDENTIAL_LABEL,
+        recipient: '98765432',
+      })
+
+    expect(res.status).toBe(400)
+    expect(res.body).toEqual({
+      message: `Demo campaign must use demo credentials. ${NON_DEMO_CREDENTIAL_LABEL} is not allowed.`,
+    })
+
+    // SecretManager should not be called
+    expect(mockGetSecretValue).not.toHaveBeenCalled()
+  })
+
+  test('Demo Campaign should be able to use demo credentials', async () => {
+    const demoCampaign = await createCampaign({ isDemo: true })
+
+    await request(app)
+      .post(`/campaign/${demoCampaign.id}/sms/credentials`)
+      .send({
+        label: DefaultCredentialName.SMS,
+        recipient: '98765432',
+      })
+
+    // Expect SecretManager to be called with default credentials
+    expect(mockGetSecretValue).toHaveBeenCalledWith(
+      formatDefaultCredentialName(DefaultCredentialName.SMS)
+    )
+  })
+})
+
 describe('PUT /campaign/{campaignId}/sms/template', () => {
+  let campaignId: number
+
+  beforeAll(async () => {
+    const campaign = await Campaign.create({
+      name: 'campaign-1',
+      userId: 1,
+      type: ChannelType.SMS,
+      valid: false,
+    })
+    campaignId = campaign.id
+  })
```

Comment:
Moving this down from above so it's clearer that it's used for template tests only